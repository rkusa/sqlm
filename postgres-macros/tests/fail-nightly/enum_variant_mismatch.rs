use postgres_types::{FromSql, ToSql};
use sqlm_postgres_macros::{sql, Enum, FromRow};

#[derive(Debug, Default, FromSql, ToSql, Enum, PartialEq, Eq)]
#[postgres(name = "role")]
enum Role {
    #[default]
    #[postgres(name = "user")]
    User,

    #[postgres(name = "moderator")]
    Moderator,
}

#[derive(Debug, PartialEq, Eq, FromRow)]
struct User {
    id: i64,
    role: Role,
}

#[tokio::main]
async fn main() {
    let _: Vec<User> = sql!("SELECT id, role FROM users").await.unwrap();
}
