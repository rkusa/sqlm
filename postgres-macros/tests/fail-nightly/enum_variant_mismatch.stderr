error[E0277]: the trait bound `Vec<User>: Query<Struct<(StructColumn<i64, "id">, StructColumn<sqlm_postgres::types::Enum<(EnumVariant<"admin">, EnumVariant<"user">)>, "role">)>>` is not satisfied
  --> tests/fail-nightly/enum_variant_mismatch.rs:23:59
   |
23 |     let _: Vec<User> = sql!("SELECT id, role FROM users").await.unwrap();
   |                                                          -^^^^^
   |                                                          ||
   |                                                          |the trait `Query<Struct<(StructColumn<i64, "id">, StructColumn<sqlm_postgres::types::Enum<(EnumVariant<"admin">, EnumVariant<"user">)>, "role">)>>` is not implemented for `Vec<User>`
   |                                                          help: remove the `.await`
   |
   = help: the trait `Query<Struct<(StructColumn<i64, "id">, StructColumn<sqlm_postgres::types::Enum<(EnumVariant<"moderator">, EnumVariant<"user">)>, "role">)>>` is implemented for `Vec<User>`
   = note: required for `Sql<'_, Struct<(StructColumn<i64, "id">, StructColumn<Enum<(EnumVariant<"admin">, ...)>, "role">)>, ...>` to implement `IntoFuture`
